<launch>
	<!-- ====================================================================== -->
	<!-- Launch file for the PATTERNGENerator                                   -->
    <!--                                                                        -->
    <!-- Launch this file, and then send messages either programmatically       -->
    <!-- or with the command:                                                   -->
    <!--                                                                        -->
    <!--    rostopic pub -1 PatternGen patterngen/MsgPattern mode shape frame hzPattern hzPoint count pointlist radius preempt   -->
    <!-- For example, a 40mm radius circle in 4 seconds, 50hz update rate.                                                       -->
    <!--    rostopic pub -1 PatternGen patterngen/MsgPattern "byshape" "circle" "Stage" 0.25 50 1 [] 40 True                     -->
    <!--                                                                        -->
    <!-- where:                                                                 -->
    <!--    mode = "byshape" or "bypoints"                                      -->
    <!--    shape = "constant" or "circle" or "square" or "flylogo" or "spiral" -->
    <!--    frame = frame of reference of the output points.                    -->
    <!--    hzPattern = pattern frequency in hertz.                             -->
    <!--    hzPoint = actuator update frequency in hertz.                       -->
    <!--    count = number of patterns to output, or -1 if forever              -->
    <!--    pointlist = [], or a list of points                                 -->
    <!--    radius = maximum distance from center                               -->
    <!--    preempt = True if this message should preempt prior messages.       -->
    <!--                                                                        -->
	<!-- ====================================================================== -->
    
	<node name="PatternGenXY" pkg="patterngen" type="PatternGenXY.py" /> 
</launch>

